src/Tools/dotnet-monitor/StartupHook/StartupHookApplicator.cs

```diff
@@ -48,14 +48,15 @@ public async Task<bool> ApplyAsync(string tfm, string fileName, CancellationToke
                 return true;
             }
 
-
-            if (_endpointInfo.RuntimeVersion.Major < 8)
+#nullable disable
+            if (_endpointInfo.RuntimeVersion?.Major < 8)
```

Comment:
Is this disablement still needed? You are guarding runtime version and altered logger  as well.

Reply:
It is required, but it's at the wrong line... good catch.

The `_logger.` call below uses `fileInfo.PhysicalPath` which may return null and should be handled in a separate PR.